WS-116: Consolidate client side and server side (#12917)
bbc/simorgh: src/app/hooks/useOptimizelyVariation/README.mdx

* Has functionality to override the Optimizely user's attributes
* Uses the `useEffect` hook to prevent issues occuring via server side rendering

> `useOptimizelyVariation` must be used in cojunction with the HOC [`withOptimizelyProvider`](https://github.com/bbc/simorgh/tree/latest/src/app/containers/PageHandlers/withOptimizelyProvider) page handler. This is because `useOptimizely` makes use of features found in the `optimizely` object client that needs to be provided by `withOptimizelyProvider`.

Reviewer: ```suggestion
`useOptimizelyVariation` must be used in conjunction with the HOC [`withOptimizelyProvider`](https://github.com/bbc/simorgh/tree/latest/src/app/containers/PageHandlers/withOptimizelyProvider) page handler. This is because `useOptimizelyVariation` makes use of features found in the `optimizely` object client that needs to be provided by `withOptimizelyProvider`.
```

`ExperimentState.CLIENT_SIDE` or  `ExperimentState.SERVER_SIDE` must be specified for the hook to call the respective client side or server side optimizely hooks.

### Usage

Here's an example of its usage:

```jsx
import withOptimizelyProvider from '#containers/PageHandlers/withOptimizelyProvider';

> import useOptimizely from '#hooks/useOptimizelyVariation';

Reviewer: ```suggestion
import useOptimizelyVariation from '#hooks/useOptimizelyVariation';
```

const ExamplePage = () => {

>   const variation = useOptimizelyVariation({ experimentName: 'foo-experiment-id', runtimeType: ExperimentState.CLIENT_SIDE});

Reviewer: ```suggestion
  const variation = useOptimizelyVariation({ experimentName: 'foo-experiment-id', experimentType: ExperimentState.CLIENT_SIDE});
```
